### PR TITLE
Add tests for RemoteMessageProcessor

### PR DESCRIPTION
**Pull request by kian-thompson**
Add unit and e2e tests for the 6 different combinations of the compression, chunking, and grouped batching options.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts [hunk 1]
```diff
@@ -85,13 +96,65 @@ const compressionSuite = (getProvider) => {
...
 			assert.strictEqual(remoteMap.get("testKey"), value);
 		});
+
+		const messageGenerationOptions = generatePairwiseOptions<{
+			compressionAndChunking?: {
+				chunking: boolean;
+			};
+			grouping: boolean;
+		}>({
+			compressionAndChunking: [undefined, { chunking: false }, { chunking: true }],
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/compression.spec.ts:114]: Quick commenting explaining what `undefined` does, and what Compression does in these three cases?

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts [hunk 2]
```diff
@@ -85,13 +96,65 @@ const compressionSuite = (getProvider) => {
...
+			grouping: [true, false],
+		});
+
+		messageGenerationOptions.forEach((option) => {
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/compression.spec.ts:118]: I'm curious if this generates all 6 combos or not.  For larger sets of options, it only makes sure each option shows up in at least one pair.

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts [hunk 3]
```diff
@@ -85,13 +96,65 @@ const compressionSuite = (getProvider) => {
...
+			await setupContainers();
 			const value = generateRandomStringOfSize(5);
 			localMap.set("testKey", value);
 
 			await provider.ensureSynchronized();
 			assert.strictEqual(localMap.get("testKey"), value);
 			assert.strictEqual(remoteMap.get("testKey"), value);
 		});
+
+		const messageGenerationOptions = generatePairwiseOptions<{
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/compression.spec.ts:108]: I know over time the _order_ these different operations happened in has fluctuated as well.  What test coverage do we have/need for different orders of operations?

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts [hunk 4]
```diff
@@ -85,13 +96,65 @@ const compressionSuite = (getProvider) => {
...
+				];
+				localDataObject.context.containerRuntime.orderSequentially(() => {
+					for (let i = 0; i < values.length; i++) {
+						localMap.set(`${i}`, values[i]);
+					}
+				});
+
+				await provider.ensureSynchronized();
+				for (let i = 0; i < values.length; i++) {
+					assert.equal(localMap.get(`${i}`), values[i]);
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/compression.spec.ts:153]: Would this actually ever break?  Is it enough to just call `ensureSynchronized`? I guess it doesn't hurt...

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts (hunk 1 above)
Comment (markfields) [reply packages/test/test-end-to-end-tests/src/test/compression.spec.ts:114]: Like, should these tests run in FullCompat mode?

**Review by markfields — APPROVED** [review]
I'll admit I didn't scrutinize line-by-line but what I did see makes sense.  And you can't merge if the tests don't pass, so...

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts (hunk 2 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/compression.spec.ts:118]: It generates all 6 combos:

```
    ✔ Correctly processes incoming messages: compression [false] chunking [false] grouping [true]
    ✔ Correctly processes incoming messages: compression [false] chunking [false] grouping [false]
    ✔ Correctly processes incoming messages: compression [true] chunking [false] grouping [true]
    ✔ Correctly processes incoming messages: compression [true] chunking [false] grouping [false]
    ✔ Correctly processes incoming messages: compression [true] chunking [true] grouping [true]
    ✔ Correctly processes incoming messages: compression [true] chunking [true] grouping [false]
```

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts (hunk 4 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/compression.spec.ts:158]: Just a simple e2e validation that combinations of compression, chunking, and op grouping don't break eventual consistency

File: packages/test/test-end-to-end-tests/src/test/compression.spec.ts (hunk 3 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/compression.spec.ts:108]: What order has changed? The only thing I can think of is grouped batching moving to be before compression/chunking, but that happened while it was an experimental feature. So, I don't think we need to test that configuration

**Pull request merged**